enumerables.rb

```diff
@@ -0,0 +1,145 @@
+module Enumerable
+  def my_each
+    count = 0
+    while count < length
+      yield(self[count])
+      count += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result << item if yield(item)
+    end
+    result
+  end
+
+  def my_all?
+    result = true
+    my_each do |item|
+      result = false unless yield(item)
+    end
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each do |item|
+      result = true if yield(item)
+    end
+    result
+  end
+
+  def my_none?
+    result = false
+    my_each do |item|
+      result = true unless yield(item)
+    end
+  end
+
+  def my_count
+    return length unless block_given?
+
+    result = 0
+    my_each do |item|
+      result = true unless yield(item)
+    end
+    result
+  end
+
+  def my_map(proc = nil)
+    result = []
+    my_each do |item|
+      result << if block_given?
+                  yield(item)
+                else
+                  proc.call(item)
+                end
+    end
+    result
+  end
+
+  def my_inject(initial = nil)
+    result = initial.nil? ? self[0] : initial
+    index = initial.nil? ? 1 : 0
+    self[index...length].my_each do |item|
+      result = yield(result, item)
+    end
+    result
+  end
```

Comment:
Your variable length is not used correctly. You meant `self.length` and maybe stickler corrected the use of it, but like this, won't work. I recommend you to convert self to an array and then iterate over that array. That way you can do what you are trying to do on line 76 and also avoid future problems in case you call `my_inject` in something other than an array.